test/loader/serializeInstructionTest.cpp

```diff
@@ -1126,4 +1242,360 @@ TEST(SerializeInstructionTest, SerializeDotProductInstruction) {
+  std::vector<uint8_t> Output;
+  std::vector<WasmEdge::AST::Instruction> Instructions;
+
+  // 19. Test convert instruction.
+  //
+  //   1.  Serialize Any__convert_extern instruction.
+  //   2.  Serialize Extern__convert_any instruction.
+  //   3.  Serialize Extern__convert_any instruction without the GC proposal.
+
+  Conf.addProposal(WasmEdge::Proposal::GC);
+  WasmEdge::AST::Instruction AnyConvertExtern(
+      WasmEdge::OpCode::Any__convert_extern);
+  WasmEdge::AST::Instruction ExternConvertAny(
+      WasmEdge::OpCode::Extern__convert_any);
+  WasmEdge::AST::Instruction End(WasmEdge::OpCode::End);
+
+  Instructions = {AnyConvertExtern, End};
+  Output = {};
+  EXPECT_TRUE(Ser.serializeSection(createCodeSec(Instructions), Output));
+  Expected = {
+      0x0AU,        // Code section
+      0x06U,        // Content size = 5
+      0x01U,        // Vector length = 1
+      0x04U,        // Code segment size = 3
+      0x00U,        // Local vec(0)
+      0xFBU, 0x1AU, // OpCode Any__convert_extern.
+      0x0BU         // Expression End.
+  };
+  EXPECT_EQ(Output, Expected);
+
+  Instructions = {ExternConvertAny, End};
+  Output = {};
+  EXPECT_TRUE(Ser.serializeSection(createCodeSec(Instructions), Output));
+  Expected[6] = 0x1BU; // OpCode Extern__convert_any.
+  EXPECT_EQ(Output, Expected);
+
+  Conf.removeProposal(WasmEdge::Proposal::GC);
+  Output = {};
+  EXPECT_FALSE(Ser.serializeSection(createCodeSec(Instructions), Output));
+}
+
+TEST(SerializeInstructionTest, SerializeI31GetInstruction) {
+  std::vector<uint8_t> Expected;
+  std::vector<uint8_t> Output;
+  std::vector<WasmEdge::AST::Instruction> Instructions;
+
+  // 20. Test I31 Get instruction.
+  //
+  //   1.  Serialize I31__get_s instruction.
+  //   2.  Serialize I31__get_u instruction.
+  //   3.  Serialize I31__get_u instruction without the GC proposal.
+
+  Conf.addProposal(WasmEdge::Proposal::GC);
+  WasmEdge::AST::Instruction I31GetS(WasmEdge::OpCode::I31__get_s);
+  WasmEdge::AST::Instruction I31GetU(WasmEdge::OpCode::I31__get_u);
+  WasmEdge::AST::Instruction End(WasmEdge::OpCode::End);
+
+  Instructions = {I31GetS, End};
+  Output = {};
+  EXPECT_TRUE(Ser.serializeSection(createCodeSec(Instructions), Output));
+  Expected = {
+      0x0AU,        // Code section
+      0x06U,        // Content size = 5
+      0x01U,        // Vector length = 1
+      0x04U,        // Code segment size = 3
+      0x00U,        // Local vec(0)
+      0xFBU, 0x1DU, // OpCode I31__get_s.
+      0x0BU         // Expression End.
+  };
+  EXPECT_EQ(Output, Expected);
+
+  Instructions = {I31GetU, End};
+  Output = {};
+  EXPECT_TRUE(Ser.serializeSection(createCodeSec(Instructions), Output));
+  Expected[6] = 0x1EU; // OpCode I31__get_u.
+  EXPECT_EQ(Output, Expected);
+
+  Conf.removeProposal(WasmEdge::Proposal::GC);
+  Output = {};
+  EXPECT_FALSE(Ser.serializeSection(createCodeSec(Instructions), Output));
+}
+
+TEST(SerializeInstructionTest, SerializeStructInstruction) {
```

Comment:
These instructions are also reference instrucitons. Combine them into the test set.